app/routes/github.js

```diff
@@ -0,0 +1,42 @@
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+const {
+  get,
+  inject: { service },
+  Route
+} = Ember;
+
+const CODE_INVALID = 'The GitHub authorization code is invalid.';
+
+export default Route.extend(AuthenticatedRouteMixin, {
+  queryParams: { code: { refreshModel: true } },
+
+  ajax: service(),
+  currentUser: service(),
+  flashMessages:service(),
+  store: service(),
+
+  model({ code }) {
+    return this._sendRequest(code);
+  },
+
+  afterModel(currentUserData) {
+    get(this, 'store').pushPayload(currentUserData);
```

Comment:
There is no need to update the currentUser in the service as I said. This ought to find the existing user record in the store and update it with new information.

Besides, the next line transitions us to `settings.profile`, which explicitly re-fetches the current user from the API.